x-pack/plugins/ml/public/application/model_management/test_models/test_trained_models_context.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createContext, Dispatch, useContext } from 'react';
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+export interface TestTrainedModelsContextType {
+  pipelineConfig: estypes.IngestPipeline | undefined;
+  createPipelineFlyoutOpen: boolean;
+}
+export const TestTrainedModelsContext = createContext<
+  | {
+      currentContext: TestTrainedModelsContextType;
+      setCurrentContext: Dispatch<TestTrainedModelsContextType>;
+    }
+  | undefined
+>(undefined);
+
+export function useTestTrainedModelsContext() {
+  const testTraindedModelsContext = useContext(TestTrainedModelsContext);
+
+  if (testTraindedModelsContext === undefined) {
+    throw new Error('TestTrainedModelsContext has not been initialized.');
+  }
+
+  return testTraindedModelsContext;
```

Comment:
I wonder if the context should be spread when returning from this function.
I'm thinking from the point of view of making it easier to use for the consumer.
As in, not having to set the whole context each time rather than just setting or reading the thing you need.
e.g.
```ts
const testTrainedModelsContext = useTestTrainedModelsContext();
testTrainedModelsContext.setCurrentContext({
  ...testTrainedModelsContext.currentContext,
  pipelineConfig: getPipelineConfig(formState),
 });
```
vs
```ts
const { setPipelineConfig } = useTestTrainedModelsContext();
setPipelineConfig(getPipelineConfig(formState));
```

or 

```ts
const testTrainedModelsContext = useTestTrainedModelsContext();

{!testTrainedModelsContext?.currentContext.createPipelineFlyoutOpen ? ( .....
```
vs
```ts
const { createPipelineFlyoutOpen } = useTestTrainedModelsContext();

{!createPipelineFlyoutOpen ? ( .....
```